src/authn/auth0.js

```diff
@@ -0,0 +1,61 @@
+import express from 'express'
+import Auth0Strategy from 'passport-auth0'
+import passport from 'passport'
+import assert from 'assert'
+import User from './../user'
+import url from 'url';
+
+class Auth0Login {
+  constructor(options) {
+    assert(options, 'options are required');
+    assert(options.cfg, 'options.cfg is required');
+    assert(options.cfg.auth0, 'options.cfg.auth0 is required');
+    assert(options.cfg.auth0.domain, 'options.cfg.auth0.domain is required');
+    assert(options.cfg.auth0.clientId, 'options.cfg.auth0.clientId is required');
+    assert(options.cfg.auth0.clientSecret, 'options.cfg.auth0.clientSecret is required');
+    assert(options.authorize, 'options.authorize is required');
+
+    this.cfg = options.cfg;
+    this.authorize = options.authorize;
+
+    passport.use(new Auth0Strategy({
+      domain: options.cfg.auth0.domain,
+      clientID: options.cfg.auth0.clientId,
+      clientSecret: options.cfg.auth0.clientSecret,
+      callbackURL: url.resolve(options.cfg.server.publicUrl, '/auth0/login'),
+    }, this.auth0Callback.bind(this)));
+  }
+
+  router() {
+    let router = new express.Router();
+
+    console.log('setting up auth0/login');
+    router.get('/login', passport.authenticate('auth0', {
+      failureRedirect: '/',
+      failureFlash: true
+    }), (req, res) => {
+      res.redirect('/');
+      return;
+    });
+
+    return router;
+  };
+
+  // for the moment, we *only* support passwordless login with Auth0.
+  auth0Callback(accessToken, refreshToken, extraParams, profile, done) {
+    try {
+      let user = new User();
+      console.log(profile._json);
+      if (!profile._json.email_verified) {
+        throw new Error('email is not verified');
+      }
+      user.identity = 'passwordless/' + profile._json.email;
```

Comment:
Will a user be able to login with `passwordless` if they have an **auth0** account for the email?

This is really `auth0-passwordless` right? Hmm... Maybe `passwordless` is fine, as we would abstract the provider.

Or perhaps this should be `email/<email>` as the only thing verified is the email?